src/constructors/lattices.c

```diff
@@ -0,0 +1,299 @@
+/*
+   IGraph library.
+   Copyright (C) 2022  The igraph development team <igraph@igraph.org>
+
+   This program is free software; you can redistribute it and/or modify
+   it under the terms of the GNU General Public License as published by
+   the Free Software Foundation; either version 2 of the License, or
+   (at your option) any later version.
+   This program is distributed in the hope that it will be useful,
+   but WITHOUT ANY WARRANTY; without even the implied warranty of
+   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+   GNU General Public License for more details.
+
+   You should have received a copy of the GNU General Public License
+   along with this program.  If not, see <https://www.gnu.org/licenses/>.
+*/
+
+#include "igraph_constructors.h"
+
+#include "core/interruption.h"
+#include "math/safe_intop.h"
+
+#define MIN(n, m) (n < m ? n : m)
+#define MAX(n, m) (n < m ? m : n)
+
+/**
+ * \ingroup generators
+ * \function igraph_triangle_lattice
```

Comment:
Please remove this `\function` line as it might interfere with the one before the primary function. Remove the `\ingroup` as well.

Reply:
Done in my commit.